enhancements/network/ovn-kubernetes-localnet-api.md

```diff
@@ -0,0 +1,557 @@
+|---------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|
+| Role                | Select the network role in the pod.                                                                                                                                                                                                                                                                   | No       |
+| PhysicalNetworkName | The OVN bridge mapping network name is configured on the node.                                                                                                                                                                                                                                        | No       |
+| MTU                 | The maximum transmission unit (MTU).                                                                                                                                                                                                                                                                  | Yes      |
+| VLAN                | The network VLAN ID.                                                                                                                                                                                                                                                                                  | Yes      |
+| Subnets             | List of CIDRs used for the pod network across the cluster.Dual-stack clusters may set 2 subnets (one for each IP family), otherwise only 1 subnet is allowed. The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `ipam.mode` is `Disabled`. | Yes      |
+| ExcludeSubnets      | List of CIDRs removed from the specified CIDRs in `subnets`.The format should match standard CIDR notation (for example, "10.128.0.0/16"). This field must be omitted if `subnetes` is unset or `ipam.mode` is `Disabled`.                                                                            | Yes      |
+| IPAM                | Contains IPAM-related configuration for the network. When ipam.lifecycle=Persistent enable workloads have persistent IP addresses. For example: Virtual Machines will have the same IP addresses along their lifecycle (stop, start migration, reboots).                                              | Yes      |
+
+#### Suggested API validations 
+- `Role`:
+  - Required.
+  - Can be Secondary only when topology=Localnet.
+    - Having Role explicitly makes the API predictable and consistent with other topologies. In addition it enables extending localnet for Primary networks.
```

Comment:
there's no use case for primary UDN in localnet. It's pointless. 

... but I understand your point.

Reply:
I changed the text to omit primary role